iphone/Maps/Categories/UIViewController+Hierarchy.swift

```diff
@@ -1,6 +1,12 @@
 extension UIViewController {
   @objc static func topViewController() -> UIViewController {
     let window = UIApplication.shared.delegate!.window!!
+    if var topController = window.rootViewController {
+      while let presentedViewController = topController.presentedViewController {
+        topController = presentedViewController
+      }
+      return topController
+    }
     return (window.rootViewController as! UINavigationController).topViewController!
```

Comment:
Why the fix above is needed? What does it fix/how to test it?
Will this line be called/is it needed or the code above will always return `topController`?

Reply:
The old implmentation doesn't return the `topController` in some cases. 

This causes a bug when the `UIAlertVieewController` cannot be shown from the modal `Menu` screen.
Is some VC1 presents VC2 modally the VC2 should be the top, not the VC1 (master's logic).

This is why we check the presentedVC property at first to check is there any modal VC's.